Login by simple saml

Shaky212 opened the pull request:
This PR provides the admin panel with authorization. 

Authorization can be given in 2 ways: 
1) via a federated login (stepup-saml) and 
2) via a local login. The federated login has yet to be tested. 

The setup is now that an admin (contact) can manage 1 or more realm configurations. A super admin (marked as super in contact database) can manage all realms. The migrations of the contact entities cannot yet be performed (and are therefore not included) due to the enum options in other entities. This has yet to be resolved.

Another issue to be resolved is the mapping between RealmContact and Realm. There seems to be an inconsistency.

Shaky212 committed:
Support both en and nl translations, default en

Shaky212 committed:
Integrate (SimpleSaml) Authorisation

This PR provides the admin panel with authorization. Authorization can be given in 2 ways:
1) via a federated login (stepup-saml) and
2) via a local login.
The federated login has yet to be tested.
The setup is now that an admin (contact) can manage 1 or more realm configurations. A super admin (marked as super in contact database) can manage all realms.
The migrations of the contact entities cannot yet be performed (and are therefore not included) due to the enum options in other entities. This has yet to be resolved.

Tjeerd commented on .gitignore:
```diff
@@ -30,3 +30,10 @@
 ###< qossmic/deptrac-shim ###
 
 /node_modules/
```
Deze staat er nu dubbel in met de introductie van webpack encore

Shaky212 replied:
check

Tjeerd commented on composer.json:
```diff
@@ -4,21 +4,24 @@
     "symfony/runtime": "6.2.*",
-    "symfony/yaml": "6.2.*",
-    "symfony/monolog-bundle": "^3.0"
+    "symfony/webpack-encore-bundle": "^1.16",
```
Denk dat je al wel doorhad dat 1 MR voor de combinatie van webpack encore en federated login niet ideaal is.... :)

Tjeerd commented on composer.json:
```diff
@@ -4,21 +4,24 @@
     "doctrine/orm": "^2.14",
     "easycorp/easyadmin-bundle": "^4.5",
     "nelmio/security-bundle": "^3.0",
+    "surfnet/stepup-saml-bundle": "dev-feature/support-sf6-dependencies",
```
normaal gesproken is het geen goed idee om deps die in `dev` status zijn te gebruiken, maar aangezien jullie deze package zelf in beheer hebben lijkt me dat hier prima

Shaky212 replied:
De SF6 support moet nog wel gereviewed en getest worden!

Tjeerd commented on config/packages/doctrine.yaml:
```diff
@@ -5,6 +5,9 @@ doctrine:
         # either here or in the DATABASE_URL env var (see .env file)
         #server_version: '15'
+        mapping_types:
+            new_enum: string
```
dit lijkt een verder niet gerelateerde wijziging, maar dit is om die ENUMs in doctrine goed te laten werken vermoed ik?

Shaky212 replied:
Klopt

Tjeerd commented on config/packages/framework.yaml:
```diff
@@ -3,7 +3,7 @@ framework:
     trusted_headers: ['x-forwarded-for', 'x-forwarded-host', 'x-forwarded-proto', 'x-forwarded-port']
     secret: '%env(APP_SECRET)%'
-    #csrf_protection: true
+    csrf_protection: true
```
In principe de default dus wijziging was niet nodig geweest, maar ach, zo is het duidelijker :)

Shaky212 replied:
✅

Tjeerd commented on config/packages/security.yaml:
```diff
@@ -5,14 +5,40 @@ security:
+                path: app_logout
+                # where to redirect after logout
+                # target: app_any_route
             # activate different ways to authenticate
```
Hierboven een aantal commented lines, bewust?

Tjeerd replied:
Zoals besproken kan alles betreft de form_login/main firewall uitgecomment met een uitleg dat dit voor dev gebruikt kan worden om gemakkelijker in te loggen.

Tjeerd commented on config/services.yaml:
```diff
@@ -24,3 +24,30 @@ services:
     # please note that last definitions always *replace* previous ones
+#    App\Security\SamlBundle\Controller\:
+#        resource: '../../Controller'
+#        tags: ['controller.service_arguments']
```
Hie wat uitgecommente regels

Shaky212 replied:
Kunnen weg vermoed ik, controller is verplaatst naar /Controller/Security/SamlBundle

Tjeerd commented on config/services.yaml:
```diff
@@ -24,3 +24,30 @@ services:
+        calls:
+            - method: setContainer
+              arguments: [ '@service_container' ]
+
```
Zijn boven- en onderstaande service definities hier echt nodig? Autowiring/Autoconfiguring is niet voldoende in dit geval?

Shaky212 replied:
Deze overname uit SPD is blijkbaar niet meer nodig, setContainer bestaat niet eens, kan dus weg.

Tjeerd replied:
Als ik https://github.com/OpenConext/Stepup-saml-bundle/blob/feature/support-sf6-dependencies/README.md lees, dan is deze controller in principe alleen nodig voor de metadata generatie

Tjeerd commented on package.json:
```diff
@@ -0,0 +1,29 @@
+        "@popperjs/core": "^2.11.6",
+        "@symfony/stimulus-bridge": "^3.2.0",
+        "@symfony/webpack-encore": "^4.0.0",
+        "bootstrap": "^5.2.3",
```
Vanwaar de dependency op bootstrap en hieronder jquery bijv?

Shaky212 replied:
Probeersel, kan weg.

Tjeerd commented on package.json:
```diff
@@ -0,0 +1,29 @@
+    },
+    "dependencies": {
+        "node-sass": "^8.0.0",
+        "sass-loader": "^13.2.0"
```
Heb je bewust gekozen om deze deps als reguliere deps te installeren itt bovenstaande dev deps?

Shaky212 replied:
Probeersel, kan weg.

Tjeerd commented on src/Command/RealmCommand.php:
```diff
@@ -0,0 +1,54 @@
+
+class RealmCommand
+{
+    public function __construct(private readonly RealmRepository $realmRepository)
```
Is dit een `Command` in de zin van een DTO om data van de ene laag in de applicatie naar een andere laag te krijgen en om de applicatie/een entity op bepaalde manier van state te laten veranderen? Zo ja, dan is de dependency op een repository 'misplaatst' :) Een DTO zou niet meer moeten zijn dan een 'classje' met wat props zonder functionaliteit.

Zo niet, dan is de naam wellicht wat verwarrend :)

Shaky212 replied:
Naam is verwarrend inderdaad, het is meer een helper-achtig iets

Tjeerd commented on src/Controller/Admin/DashboardController.php:
```diff
@@ -14,9 +21,32 @@
 class DashboardController extends AbstractDashboardController
 {
-    #[Route('/admin', name: 'admin')]
+    /** @throws Exception */
```
Wat mij betreft is dit overbodig. Voegt verder niets toe toch?

Tjeerd replied:
Komt op nog een aantal andere plaatsen ook terug

Tjeerd commented on src/Controller/Admin/DashboardController.php:
```diff
@@ -2,8 +2,15 @@
+
 namespace App\Controller\Admin;
 
+use _PHPStan_1f608dc6a\Nette\Neon\Exception;
```
Dat lijkt me niet de juiste `use`.

Shaky212 replied:
Nee, moet Exception zijn.

Tjeerd commented on src/Controller/Admin/DashboardController.php:
```diff
@@ -14,9 +21,32 @@
+    public function indexDefaultLocale(): Response
+    {
+        if (!($this->isGranted('ROLE_SUPER_ADMIN') || $this->isGranted('ROLE_ADMIN'))) {
+            throw new Exception('You have no access to this resource');
```
Iets netter en makkelijker: `throw $this->createAccessDeniedException();`

Shaky212 replied:
Veel netter idd

Tjeerd commented on src/Controller/Admin/DashboardController.php:
```diff
@@ -14,9 +21,32 @@
+    #[Route('/admin', name: 'admin_default_locale')]
+    public function indexDefaultLocale(): Response
+    {
+        if (!($this->isGranted('ROLE_SUPER_ADMIN') || $this->isGranted('ROLE_ADMIN'))) {
```
Als ik het goed gezien heb dan is er een hiërarchische relatie tussen super_admin en admin, dus alleen een check voor admin zou voldoende moeten zijn.

Shaky212 replied:
Klopt, aanpassen...

Tjeerd commented on src/Controller/Admin/DashboardController.php:
```diff
@@ -14,9 +21,32 @@
+    }
+
+    /** @throws Exception */
+    #[Route('/admin/{_locale}', name: 'admin')]
```
Is deze route expliciet nog extra nodig naast bovenstaande? _locale lijkt optioneel toch?

Shaky212 replied:
In het admin menu kun je switchen van taal wisselen, maar daar wordt _locale als queryparameter gebruikt, route kan weg.

Tjeerd commented on src/Controller/Admin/DashboardController.php:
```diff
@@ -14,9 +21,32 @@
+        if (!($this->isGranted('ROLE_SUPER_ADMIN') || $this->isGranted('ROLE_ADMIN'))) {
+            throw new Exception('You have no access to this resource');
+        }
+
```
Overigens, heb je in security.yaml deze routes al afgeschermd, dus vraag me af of deze handmatige checks hier uberhaupt benodigd zijn.

Shaky212 replied:
klopt

Tjeerd commented on src/Controller/Admin/RealmSigningLogCrudController.php:
```diff
@@ -50,8 +72,14 @@ public function configureFields(string $pageName): iterable
+                }),
+            TextField::new('client')
+                ->formatValue(static function ($value, $entity) {
+                    return $value ? $value : '-';
```
`return $value ?? '-';`

Shaky212 replied:
Yep, is wat korter.

Tjeerd commented on src/Controller/Admin/RealmSigningLogCrudController.php:
```diff
@@ -94,64 +122,68 @@ public function configureAssets(Assets $assets): Assets
     }
+
+    /** @return array<Realm> */
+    public function getRealmsChoicesOfUser(): array
```
`private` en wellicht de hele functie binnen de service class plaatsen? Dat kan je hem nog eens hergebruiken naast dat dit geen controller logica is, toch?

Tjeerd commented on src/Controller/SecurityController.php:
```diff
@@ -0,0 +1,42 @@
+    {
+        // if ($this->getUser()) {
+        //     return $this->redirectToRoute('target_path');
+        // }
```
commented lines

Shaky212 replied:
Kennelijk al aangepast.

Tjeerd commented on src/Controller/Admin/RealmSigningLogCrudController.php:
```diff
@@ -94,64 +122,68 @@ public function configureAssets(Assets $assets): Assets
+            return $queryBuilder;
+        }
+
+        $user = $this->tokenStorage->getToken()->getUser();
```
Ik meen dat er een `getUser` convenience method is in de abstractController.

Shaky212 replied:
Dit is inderdaad het geval!

Tjeerd commented on src/Controller/Admin/RealmSigningLogCrudController.php:
```diff
@@ -94,64 +122,68 @@ public function configureAssets(Assets $assets): Assets
-            throw new Exception($e->getMessage());
-        }
+        $entity = $context->getEntity()->getInstance();
+        $this->denyAccessUnlessGranted('edit', $entity);
```
Is het niet per ongeluk al mogelijk om de required permission bij de action configuration hierboven aan te geven? Geen idee, maar dat zou me een mooie optie lijken :)

Shaky212 replied:
Heb even gekeken, maar zag die mogelijk niet.

Tjeerd commented on src/Entity/Contact.php:
```diff
@@ -0,0 +1,158 @@
+    private string $password;
+
+    #[ORM\Column]
+    private bool $super;
```
Wat is `super`  ?

Tjeerd replied:
Na verder lezen ben ik erachter dat dit SuperAdmin betekent... dan zou ik ook zo noemen :)

Tjeerd commented on src/Entity/Contact.php:
```diff
@@ -0,0 +1,158 @@
+
+    public function eraseCredentials(): void
+    {
+        // TODO: Implement eraseCredentials() method.
```
todo

Tjeerd commented on src/Entity/Contact.php:
```diff
@@ -0,0 +1,158 @@
+    {
+        $realmContacts = $this->getRealmContacts();
+        foreach ($realmContacts as $realmContact) {
+            if ($realm->getRealm() === $realmContact->getRealm()->getRealm()) {
```
Gaat dit goed met een strict comparison op objecten? Is het mss slimmer om de primary key te vergelijken?

Shaky212 replied:
(get)Realm is de primary

Tjeerd commented on src/Entity/Contact.php:
```diff
@@ -0,0 +1,158 @@
+    public function getRoles(): array
+    {
+        if (count($this->roles) === 0) {
+            $this->getSuper() ? $this->roles[] = 'ROLE_SUPER_ADMIN' :  $this->roles[] = 'ROLE_ADMIN';
```
Dus je bent altijd admin of super_admin?

Tjeerd replied:
Overigens zou ik het eerder schrijven als: 
`$this->roles[] = $this->super ? 'ROLE_SUPER_ADMIN': 'ROLE_ADMIN'; `

Shaky212 replied:
Mee , suffe constructie van mijn kant.

Tjeerd commented on src/Entity/Contact.php:
```diff
@@ -0,0 +1,158 @@
+        return $this;
+    }
+
+    public function getSuper(): bool
```
`isSuperAdmin` ?

Shaky212 replied:
isSuperAmin of superAdmin is duidelijker idd

Tjeerd commented on src/Entity/Realm.php:
```diff
@@ -0,0 +1,52 @@
+
+    /** @var Collection<RealmContact> */
+    #[ORM\OneToMany(mappedBy: 'contact', targetEntity: RealmContact::class)]
+    #[ORM\JoinColumn(name: 'realm', referencedColumnName: 'realm', nullable: false)]
```
Volgens mij kan je bij een OneToMany niets over de JoinColumn zeggen, maar doe je dat aan de andere kant van de relatie.

Shaky212 replied:
Klopt, zie volgende.

Tjeerd commented on src/Entity/Realm.php:
```diff
@@ -0,0 +1,52 @@
+    private string|null $realm = null;
+
+    /** @var Collection<RealmContact> */
+    #[ORM\OneToMany(mappedBy: 'contact', targetEntity: RealmContact::class)]
```
Zou deze niet mappedBy `realm` moeten zijn?

Shaky212 replied:
Klopt, met bovenstaande is daarmee de laatste schema:validation gefixed

Tjeerd commented:
Ben nog niet klaar met de review, ga er later weer mee verder.

Tjeerd commented on src/Entity/RealmContact.php:
```diff
@@ -0,0 +1,65 @@
+    public function __construct()
+    {
+        $this->contact = new Contact();
+        $this->realm   = new Realm();
```
Dat lijkt me gevaarlijk om zo te instantiëren. Die entities bestaan dan wel, maar zijn verder "leeg".

Shaky212 replied:
Auto generation left-over

Tjeerd commented on src/Entity/RealmSigningLog.php:
```diff
@@ -65,17 +68,21 @@ class RealmSigningLog
     #[ORM\Column(length: 39, nullable: true)]
     private string|null $ip = null;
 
+    #[ORM\ManyToOne]
```
mss voor de zekerheid, ondanks dat het niet perse hoeft denk ik, toch maar weer even de targetEntity en de inversed_by neerzetten

Shaky212 replied:
Bewust niet gedaan, omdat een realm 100k+ logs kan bevatten.

Tjeerd commented on src/Security/Authenticator/LoginFormAuthenticator.php:
```diff
@@ -0,0 +1,76 @@
+use Symfony\Component\Security\Http\Authenticator\Passport\Passport;
+use Symfony\Component\Security\Http\Util\TargetPathTrait;
+
+class LoginFormAuthenticator extends AbstractLoginFormAuthenticator
```
Is dit nodig voor saml auth? Voor zover ik weet iig niet voor de reguliere form login

Tjeerd replied:
Als dat zo is, zou ik het wellicht allemaal samen in een eigen directory/namespace stoppen

Shaky212 replied:
Dit is voor de lokale login. Inloggen via simple saml zit in  in 'Samlbundle'. Wellicht is het niet duidelijk en toch een aparte map?

Tjeerd replied:
Deze kan weg denk ik

Shaky212 replied:
Klopt, was een euke vingeroefening maar symfony kan dit zelf.

Tjeerd commented on src/Security/Identity.php:
```diff
@@ -0,0 +1,51 @@
+use App\Entity\Contact;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class Identity implements UserInterface
```
Zoals hierboven, als dit voor de saml integratie si, wellicht even in een eigen namespace.

Tjeerd commented on src/Security/SamlBundle/DashboardSamlBundle.php:
```diff
@@ -0,0 +1,22 @@
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+class DashboardSamlBundle extends Bundle
```
Vermoed dat deze file wegkan, je hebt dit al in de vendors ergens staan gok ik :)

Tjeerd replied:
Eigenlijk alles wat onder SamlBundle staat kan weg denk ik

Shaky212 replied:
Dit is het stukje implementatie client side en nodig voor authenticatie.

Tjeerd replied:
Uiteindelijk heb je alleen de provider een een controller voor je metadata nodig kwamen we gisteren achter.

Shaky212 replied:
Klopt, haal m maar meteen weg.

Tjeerd commented on src/Security/Voter/RealmSigningLogVoter.php:
```diff
@@ -0,0 +1,52 @@
+    protected function supports(string $attribute, mixed $subject): bool
+    {
+        // if the attribute isn't one we support, return false
+        return $attribute === self::EDIT;
```
Wellicht wel even checken of subject ook een RealmSigningLog is, anders komt elke security check hierlangs.

Shaky212 replied:
Ok

Tjeerd commented on src/Security/Voter/RealmSigningLogVoter.php:
```diff
@@ -0,0 +1,52 @@
+            return false;
+        }
+
+        $realmSigningLog = $subject;
```
Deze assignment lijkt me overbodig :)

Shaky212 replied:
Kan weg idd.

Tjeerd commented on templates/security/login.html.twig:
```diff
@@ -0,0 +1,33 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}Log in!{% endblock %}
```
Deze nog even door translate halen

Shaky212 replied:
Yep, vooralsnog maar doen, maar ik vermoed wel dat hier iets anders komt te staan

Tjeerd commented on webpack.config.js:
```diff
@@ -0,0 +1,76 @@
+    .enableSassLoader()
+
+    // uncomment if you use TypeScript
+    //.enableTypeScriptLoader()
```
Deze wellicht nog aan zetten (en gebruiken dan natuurlijk) :)

Tjeerd commented on templates/security/login.html.twig:
```diff
@@ -0,0 +1,33 @@
+
+    {% if app.user %}
+        <div class="mb-3">
+            You are logged in as {{ app.user.userIdentifier }}, <a href="{{ path('app_logout') }}">Logout</a>
```
Hier nog een translate block omheen

Shaky212 replied:
Is in een latere PR gedaan

Shaky212 committed:
Edit realm configuration

This is the first setup of the realm configuration. Related entities are created and maintained.
Also a raw crud for the contacts is added

Shaky212 committed:
Move items for tranlation

Shaky212 committed:
Clean up unused items

Shaky212 committed:
Improve naming and namespace adjustment

Shaky212 committed:
Validate relation scheme

Shaky212 committed:
clean up double/unused references

Shaky212 committed:
Repair after rebase

Shaky212 committed:
Revoke user accounts

We chose to place a realm_siging_user view on the realm_signing_log entity. In this way, aggregated values can be made transparent via easy_admin. The final actions are always done on the realm_singing_log table.

Shaky212 committed:
Clear dev-left overs and rename methods

Shaky212 committed:
Show totals of pseudo and user accounts

Shaky212 committed:
Repair parameter type to pass the correct object

Shaky212 committed:
Add password and password repeat to the contact crud

Shaky212 committed:
Add crud to manage realms and administrators

Shaky212 committed:
Handle the federal login

The test IDP host test.surfconext.nl is used to test the federal login. Finally, an authorized user presented to the Authenticator is validated within the module and redirected to the admin panel.

Shaky212 committed:
Add network profiles to the realm

After running the demo, some changes have been suggested for editing the realms. These are broadly processed, including the network profiles, helpdesk, vhost are linked to a realm.

Tjeerd commented on config/packages/security.yaml:
```diff
@@ -5,14 +5,40 @@ security:
             security: false
+#
+#        login_firewall:
+#                pattern: ^/saml/metadata
```
Dit kan vermoedelijk weg, in principe is je hele applicatie afgeschermd door main/saml_based firewall.

Shaky212 replied:
Ik laat het nog even staan. Bij een werkende federatieve login kan dit gecontroleerd worden.

Tjeerd commented on config/packages/security.yaml:
```diff
@@ -5,14 +5,40 @@ security:
+        app_user_provider:
+            entity:
+                class: App\Entity\Contact
+                property: email
```
Ook dit kan uitgecomment omdat het alleen gebruikt wordt door de form_login method.

Tjeerd replied:
Je mist hier dan nog wel de saml-provider

Shaky212 replied:
Is in een latere pr opgenomen.

Tjeerd commented on config/packages/security.yaml:
```diff
@@ -5,14 +5,40 @@ security:
+#                login_path: app_login
+#                check_path: app_login
+#                enable_csrf: true
+            custom_authenticator: App\Security\Authenticator\LoginFormAuthenticator
```
Zoals besproken lijkt er geen noodzaak voor een custom authenticator.

Shaky212 replied:
Kan weg idd

Tjeerd commented on config/packages/security.yaml:
```diff
@@ -22,7 +48,7 @@ security:
     # Note: Only the *first* access control that matches will be used
     access_control:
-        # - { path: ^/admin, roles: ROLE_ADMIN }
+        - { path: ^/admin, roles: ROLE_ADMIN }
```
Wellicht nog een expliciete config opnemen voor de /saml/metadata route?

Shaky212 replied:
Op een later moment opnemen en testen of dit werkt bij de saml_based login

Tjeerd commented on config/services.yaml:
```diff
@@ -24,3 +24,22 @@ services:
 
     # add more service definitions when explicit configuration is needed
     # please note that last definitions always *replace* previous ones
+    App\Security\Authenticator\LoginFormAuthenticator:
```
Deze lijkt dus niet nodig

Tjeerd commented on config/services.yaml:
```diff
@@ -24,3 +24,22 @@ services:
+
+    # Firewall
+    surfnet_saml.saml_provider:
+        class: App\Security\SamlBundle\Authentication\SamlProvider
```
Denk dat je de expliciete config voor de provider weg kan laten, als je de alias hieronder maar wel laat staan.

Tjeerd commented on config/services.yaml:
```diff
@@ -24,3 +24,22 @@ services:
+        class: App\Security\SamlBundle\Authentication\SamlProvider
+
+    App\Security\SamlBundle\Authentication\SamlProvider:
+        alias: surfnet_saml.saml_provider
```
Ik gok dat dit iets korter kan: `surfnet_saml.saml_provider: '@App\Security\SamlBundle\Authentication\SamlProvider`.

Overigens zou ik er niet voor kiezen om een "SamlBundle" namespace te introduceren, dat is wellicht verwarrend omdat je hier niet een Bundle aan het bouwen bent. Ik zou dat stukje gewoon weglaten denk ik.

Tjeerd commented on src/Controller/Admin/DashboardController.php:
```diff
@@ -14,9 +21,32 @@
     {
+        if (!($this->isGranted('ROLE_SUPER_ADMIN') || $this->isGranted('ROLE_ADMIN'))) {
+            throw new Exception('You have no access to this resource');
+        }
```
Zie opmerking hierboven

Tjeerd commented on src/Controller/SecurityController.php:
```diff
@@ -0,0 +1,42 @@
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Http\Authentication\AuthenticationUtils;
+
+class SecurityController extends AbstractController
```
Deze dus alleen als je lokaal zonder saml wilt inloggen. Levert dit echter geen security risk op als je hem zo laat staan?

Shaky212 replied:
Bij testen saml_based login checken

Tjeerd commented on src/Controller/Security/SamlBundle/SamlController.php:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    #[Route('/saml/metadata', name:'dashboard_saml_metadata', methods: ['GET'])]
+    public function metadataAction(): XMLResponse
```
Alleen deze method in de controller zou voldoende moeten zijn

Shaky212 replied:
Bij testen saml_based login checken

Shaky212 committed:
Simplify local login process

After running the demo, some changes have been suggested for editing the realms. These are broadly processed, including the network profiles, helpdesk, vhost are linked to a realm.

Shaky212 committed:
Merge branch 'feature/login-by-simple-saml' into feature/manage-realm-configuration

Shaky212 committed:
Move or remove dependencies

Shaky212 committed:
Translate the untranslated

Shaky212 committed:
Remove dev-leftover

Shaky212 committed:
Remove dev-leftover

Shaky212 committed:
Merge remote-tracking branch 'origin/feature/manage-realm-configuration' into feature/manage-realm-configuration

Shaky212 committed:
Improve accessibility checks

Shaky212 committed:
Improve accessibility checks

Shaky212 committed:
Refactor commands and other review issues

Shaky212 committed:
Merge pull request #10 from geteduroam/feature/edit-realm-configuration

Edit realm configuration

Shaky212 committed:
Merge pull request #9 from geteduroam/feature/assign-realms-to-admins

Assign realms to admins

Shaky212 committed:
Use symfony csrf defaults

Shaky212 committed:
Introduce application layer

Shaky212 committed:
Merge branch 'feature/login-by-simple-saml' into feature/manage-realm-configuration

Shaky212 committed:
Merge branch 'feature/manage-realm-configuration' into feature/revoke-users

Shaky212 committed:
Merge pull request #7 from geteduroam/feature/revoke-users

Revoke users

Shaky212 committed:
Merge pull request #6 from geteduroam/feature/manage-realm-configuration

Manage realm configuration

Shaky212 committed:
Merge branch 'feature/dashboard-adjustments' into feature/revoke-users

Shaky212 committed:
Merge branch 'feature/revoke-users' into feature/manage-realm-configuration

Shaky212 committed:
Merge branch 'feature/manage-realm-configuration' into feature/login-by-simple-saml

Shaky212 committed:
Resolve iterative merge problems

Shaky212 committed:
Resolve iterative merge problems

Shaky212 committed:
Merge branch 'feature/login-by-simple-saml' of github.com:geteduroam/letswifi-admin into feature/login-by-simple-saml

Shaky212 committed:
Repair method name to save realms

Shaky212 committed:
Feedback
